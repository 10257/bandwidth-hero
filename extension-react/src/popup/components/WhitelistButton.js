import React from 'react'
import { Segment, Button } from 'semantic-ui-react'

export default () => {
  return (
    <Segment attached>
      <Button content="Whitelist This Site" basic positive fluid />
    </Segment>
  )
}
