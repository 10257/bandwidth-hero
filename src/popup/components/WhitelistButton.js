import React from 'react'
import { Segment, Button } from 'semantic-ui-react'
import parseUrl from '../../utils/parseUrl'

export default ({
  whitelist,
  currentUrl,
  onAddToWhitelist,
  onRemoveFromWhitelist
}: {
  whitelist: string[],
  currentUrl: string,
  onAddToWhiteList: () => void,
  onRemoveFromWhitelist: () => void
}) => {
  const { schema, hostname } = parseUrl(currentUrl)

  if (!/^https?:/i.test(schema)) return null
  if (whitelist.includes(hostname)) {
    return (
      <Segment attached>
        <Button
          content="Remove from whitelist"
          onClick={onRemoveFromWhitelist}
          basic
          negative
          fluid
        />
      </Segment>
    )
  } else {
    return (
      <Segment attached>
        <Button
          content="Whitelist this site"
          onClick={onAddToWhitelist}
          basic
          positive
          fluid
        />
      </Segment>
    )
  }
}
