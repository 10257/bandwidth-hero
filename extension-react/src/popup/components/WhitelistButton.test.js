import React from 'react'
import WhitelistButton from './WhitelistButton'
import renderer from 'react-test-renderer'

it('renders whitelist button correctly', () => {
  const tree = renderer
    .create(
      <WhitelistButton whitelist={[]} currentUrl="https://google.com/foo.png" />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders remove from whitelist button correctly', () => {
  const tree = renderer
    .create(
      <WhitelistButton
        whitelist={['google.com']}
        currentUrl="https://google.com/foo.png"
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('does not render on non-URL pages', () => {
  const tree = renderer
    .create(<WhitelistButton whitelist={[]} currentUrl="file:///foo.png" />)
    .toJSON()
  expect(tree).toBeNull()
})
