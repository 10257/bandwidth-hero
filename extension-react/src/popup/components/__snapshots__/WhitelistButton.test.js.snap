// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders remove from whitelist button correctly 1`] = `
<div
  className="ui attached segment"
>
  <button
    className="ui basic fluid negative button"
    onClick={[Function]}
    tabIndex={undefined}
  >
    Remove from whitelist
  </button>
</div>
`;

exports[`renders whitelist button correctly 1`] = `
<div
  className="ui attached segment"
>
  <button
    className="ui basic fluid positive button"
    onClick={[Function]}
    tabIndex={undefined}
  >
    Whitelist this site
  </button>
</div>
`;
